### PR TITLE
Create social media navigation

### PR DESCRIPTION
**Pull request by jiabaow**
add a menu bar
create empty Follows/Profile/Search pages

### CODE REVIEW

File: fe1-web/parts/lao/socialMedia/SocialMediaNavigation.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+        if (route.name === STRINGS.social_media_navigation_tab_home) {
+          iconName = 'home';
+        } else if (route.name === STRINGS.social_media_navigation_tab_search) {
+          iconName = 'search';
+        } else if (route.name === STRINGS.social_media_navigation_tab_follows) {
+          iconName = 'people';
+        } else if (route.name === STRINGS.social_media_navigation_tab_profile) {
+          iconName = 'person';
+        }
```
Comment (pierluca) [inline fe1-web/parts/lao/socialMedia/SocialMediaNavigation.tsx:27]: is it me or a `switch` might seem cleaner? :-)

File: fe1-web/parts/lao/socialMedia/SocialMediaNavigation.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            break;
+          default:
+            console.log('wrong route.');
+            break;
+        }
+
+        return <Ionicons name={iconName} size={23} color={color} />;
+      },
+      tabBarActiveTintColor: 'tomato',
+      tabBarInactiveTintColor: 'gray',
```
Comment (Xelowak) [inline fe1-web/parts/lao/socialMedia/SocialMediaNavigation.tsx:40]: Maybe you can add these in `styles/colors.js`. We already have a gray there btw.

File: fe1-web/parts/lao/socialMedia/SocialProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { Text, View } from 'react-native';
+
+const SocialProfile = () => (
+  <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+    <Text>My Profile</Text>
```
Comment (Xelowak) [inline fe1-web/parts/lao/socialMedia/SocialProfile.tsx:6]: I think it's better to use the string you created in STRINGS.

File: fe1-web/parts/lao/socialMedia/SocialSearch.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { Text, View } from 'react-native';
+
+const SocialSearch = () => (
+  <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+    <Text>Search</Text>
```
Comment (Xelowak) [inline fe1-web/parts/lao/socialMedia/SocialSearch.tsx:6]: Same comment.

File: fe1-web/parts/lao/socialMedia/SocialFollows.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { Text, View } from 'react-native';
+
+const SocialFollows = () => (
+  <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+    <Text>My Follows</Text>
```
Comment (Xelowak) [inline fe1-web/parts/lao/socialMedia/SocialFollows.tsx:6]: I think it's better to use the string you created in STRINGS.

File: fe1-web/tsconfig.json [hunk 6]
```diff
@@ -29,7 +30,7 @@
     "esModuleInterop": true,                  /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */
     "resolveJsonModule": true,                /* Enables importing JSON modules */
     "skipLibCheck": true,                     /* Skip type checking of declaration files. */
-    "forceConsistentCasingInFileNames": true, /* Disallow inconsistently-cased references to the same file. */
+    "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */
```
Comment (Xelowak) [inline fe1-web/tsconfig.json:33]: I think we can keep this trailing comma.

**Review by Xelowak — APPROVED** [review]
Good job ! I made some small comments you might need consider, but that's nice overall.

File: fe1-web/parts/lao/socialMedia/SocialProfile.tsx (hunk 3 above)
Comment (jiabaow) [reply fe1-web/parts/lao/socialMedia/SocialProfile.tsx:6]: This is just an example and will be deleted for sure after finishing this page, but yes, I can change it.

File: fe1-web/jest/setup.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import 'react-native-gesture-handler/jestSetup';
+import { jest } from '@jest/globals';
+
+jest.mock('react-native-reanimated', () => {
+  // eslint-disable-next-line global-require
+  const Reanimated = require('react-native-reanimated/mock');
+
+  // The mock for `call` immediately calls the callback which is incorrect
+  // So we override it with a no-op
+  Reanimated.default.call = () => {};
```
Comment (pierluca) [inline fe1-web/jest/setup.js:10]: Sonar is complaining about this.
I'd suggest fixing it in the simplest way, by adding a comment in the arrow function to clarify the intent.

```suggestion
  Reanimated.default.call = () => { /* no-op */ };
```

**Review by pierluca — APPROVED** [review]
Looks good to me, but do fix Sonar's complaint :-)

**Comment by pierluca** [comment]
Tiny remark: try to avoid `...` as a commit message 😉

**Comment by jiabaow** [comment]
> Tiny remark: try to avoid `...` as a commit message and use something more informative😉

I wrote this when I switched to another branch but the current modification is not done yet. I will be more careful with this next time!

File: fe1-web/package.json [hunk 8]
```diff
@@ -56,6 +58,7 @@
     "react-navigation": "^4.4.1",
     "react-qr-reader": "^2.2.1",
     "react-redux": "^7.2.4",
+    "react-test-renderer": "^17.0.2",
```
Comment (pierluca) [inline fe1-web/package.json:61]: I assume you don't need the react-test-renderer, right ? 🤔

**Review by pierluca — COMMENTED** [review]
The `package.json` still seems to have spurious dependencies.
I can force the merge once that's fixed.

Also: I understand the need to commit work in progress, a common acronym to indicate that is "WIP" (work-in-progress). But usually WIP commits are then squashed together into a single commit to keep a cleaner Git history :-)

**Pull request closed without merging**